aidial_adapter_openai/gpt4_multi_modal/chat_completion.py

```diff
@@ -180,26 +201,45 @@ async def chat_completion(
 
     api_url = f"{upstream_endpoint}?api-version={api_version}"
 
-    result = await transform_messages(file_storage, messages)
-
-    if isinstance(result, str):
-        logger.error(f"Failed to prepare request: {result}")
+    transformation_result = await transform_messages(
+        file_storage, messages, tokenizer
+    )
 
+    if isinstance(transformation_result, DialException):
+        logger.error(
+            f"Failed to prepare request: {transformation_result.message}"
+        )
         chunk = create_stage_chunk("Usage", USAGE, is_stream)
-
-        exc = InvalidRequestError(message=result, display_message=result)
-
-        return create_response_from_chunk(chunk, exc, is_stream)
-
-    new_messages, prompt_image_tokens = result
-
-    prompt_text_tokens = tokenizer.calculate_prompt_tokens(messages)
-    estimated_prompt_tokens = prompt_text_tokens + prompt_image_tokens
+        return create_response_from_chunk(
+            chunk, transformation_result, is_stream
+        )
+    max_prompt_tokens = None
+    if "max_prompt_tokens" in request:
+        max_prompt_tokens = request.pop("max_prompt_tokens")
```

Comment:
Could be simplified to `max_prompt_tokens = request.pop("max_prompt_tokens", None)`

Reply:
fixed